commands/update.py

```diff
@@ -45,16 +45,27 @@ def do_update(args):
         logging.info("The latest available version is %s and you have version %s" % (new_version, version))
         logging.debug("comparing %s to %s", version, new_version)
         perform_update = (semver.compare(version, new_version) < 0)
+        url = board.get_download_url(new_version, board_id, 'uf2', args.locale)
     else:
         # Forcing an update to a particular version
 
         new_version = args.firmware_version
         perform_update = True
+        url = board.get_download_url(new_version, board_id, 'uf2', args.locale)
+
+    if args.commit_hash is not None:
+        s3_path = urlutil.find_firmware_by_hash(args.commit_hash, board_id, args.locale)
+        if s3_path:
+            perform_update = True
+            url = urlutil.get_s3_url(s3_path)
+            new_version = args.commit_hash
+        else:
+            print("")
```

Comment:
I think you want something else here. Like some kind of log.critical and an exception thrown. You can probably find another example of this somewhere else, but I think if you hit this we will definitely not be going to space today.

Reply:
Whoops, thank you. critical log added in latest commit. I think it's okay to just print and not throw the exception but lmk if you want it to raise in addition.